packages/autorest.typescript/src/restLevelClient/transforms/transformResponseTypes.ts

```diff
@@ -0,0 +1,144 @@
+import { CodeModel, HttpHeader, Operation, Response, SchemaContext, SchemaResponse } from "@autorest/codemodel";
+import { HeaderMetadata, ImportKind, OperationResponse, ResponseMetadata } from "@azure-tools/rlc-codegen";
+import { getLanguageMetadata } from "../../utils/languageHelpers";
+import { responseToSchemaResponse } from "../operationHelpers";
+import { getElementType, getFormatDocs, primitiveSchemaToType } from "../schemaHelpers";
+
+export function transformResponseTypes(model: CodeModel, importDetails: Map<ImportKind, Set<string>>): OperationResponse[] {
+    const operations = getAllOperationRequests(model);
+    const rlcResponses: OperationResponse[] = [];
+    let importedModels = new Set<string>();
+    for (const operation of operations) {
+        const responses = mergeResponsesAndExceptions(operation);
+        const operationLanguageMetadata = getLanguageMetadata(operation.language);
+        const rlcOperationUnit: OperationResponse = {
+            operationName: `${operationLanguageMetadata.name}`,
+            responses: []
+        };
+        for (const response of responses) {
+            // We'll need http information such as headers, statusCodes, etc continue if not available
+            if (!response.protocol.http) {
+                continue;
+            }
+            let schemaResponse: SchemaResponse = responseToSchemaResponse(response);
+            const rlcResponseUnit: ResponseMetadata = {
+                statusCode: getStatusCode(schemaResponse),
+                description: operationLanguageMetadata.description
+            };
+
+            // transform header
+            transformHeaders(response, rlcResponseUnit);
+            // transform body
+            transformBody(response, importedModels, rlcResponseUnit);
+            rlcOperationUnit.responses.push(rlcResponseUnit);
+        }
+        rlcResponses.push(rlcOperationUnit);
+    }
+    if (importedModels.size > 0) {
+        importDetails.set(ImportKind.ResponseOutput, importedModels);
+    }
+    return rlcResponses;
+}
+
+
+function transformHeaders(from: Response, to: ResponseMetadata) {
```

Comment:
I'd suggest to make this function return the ResponseMetadata instead of populating it in the parameter `to`.


```ts
function transformHeaders(from: Response, to: ResponseMetadata)
```

And use it
```ts
const headers = transformHeaders(response, rlcResponseUnit);
const body = transformBody(response)

rlcResponseUnit = {
...rlcResponseUnit,
headers,
body
}
```

Pure functions are preferred as they are easier to test and more predictable when they don't cause side effects. So rule of thumb is, whenever you can return and not modify parameters we should do that, just modify parameters when there is no other option

Reply:
This is one of the reasons I want to refactor at some point the functions that take the importedModels, importedParameters and clientImports sets